src/nodes.coffee

```diff
@@ -1200,6 +1186,35 @@ exports.Value = class Value extends Base
     else
       @error 'tried to assign to unassignable value'
 
+  ast: ->
+    # If the `Value` has no properties, the AST node is just whatever this
+    # node’s `base` is.
+    return @base.ast() unless @hasProperties()
+    super()
+
+  astType: -> 'MemberExpression'
+
+  # If this `Value` has properties, the *last* property (e.g. `c` in `a.b.c`)
+  # becomes the `property`, and the preceding properties (e.g. `a.b`) become
+  # a child `Value` node assigned to the `object` property.
+  astProperties: ->
+    [initialProps..., property] = @properties
+
+    object: new Value(@base, initialProps, @tag, @isDefaultValue).ast()
+    property: property.ast()
+    computed: property instanceof Index or property.name?.unwrap() not instanceof PropertyName
+    optional: !!property.soak
+    shorthand: !!property.shorthand
+
+  # When the `Value` has properties, the location data of a
+  # `MemberExpression` AST node corresponding to a given property should
+  # span the location of the `Value`’s `base` (including parentheses if
+  # present) through the property’s location.
+  astLocationData: ->
```

Comment:
I think this is a good idea. I went ahead and implemented `mergeLocationData` and this new helper method you’re describing, and it works as you predicted. I called the helper method `object` rather than `withoutLastProperty`, since `object` is the name of the AST field this output is going into, but I’m not entirely sure what the best name is. I see the point of `withoutLastProperty` as being “look, this is a general helper that doesn’t involve AST stuff” yet it _does_ involve AST generation in that we only need this helper for AST stuff. So I dunno about the name, but the actual code works great 😄